auth0/src/test/java/com/auth0/android/request/DefaultClientTest.kt

```diff
@@ -48,6 +51,59 @@ public class DefaultClientTest {
         mockServer.shutdown()
     }
 
+    @Test
+    public fun shouldAddDefaultHeadersToRequests() {
+        enqueueMockResponse(STATUS_SUCCESS, JSON_OK)
+        executeRequest(
+            HttpMethod.GET,
+            DefaultClient(defaultHeaders = mapOf("custom-header" to "custom-value"))
+        )
+
+        val sentRequest = mockServer.takeRequest()
+        requestAssertions(sentRequest, HttpMethod.GET, mapOf("custom-header" to "custom-value"))
+    }
+
+    @Test
+    public fun shouldOverrideDefaultHeadersWithRequestHeadersIfSameKey() {
+        enqueueMockResponse(STATUS_SUCCESS, JSON_OK)
+
+        executeRequest(
+            HttpMethod.GET,
+            DefaultClient(defaultHeaders = mapOf("a-header" to "a-value")),
+            requestHeaders = mutableMapOf("a-header" to "updated-value"))
+
+        val sentRequest = mockServer.takeRequest()
+        requestAssertions(sentRequest, HttpMethod.GET, mapOf("a-header" to "updated-value"))
+    }
+
+    @Test
+    public fun shouldHaveLoggingDisabledByDefault() {
+        assertThat(DefaultClient().okHttpClient.interceptors, empty())
+    }
+
+    @Test
+    public fun shouldHaveLoggingEnabledIfSpecified() {
+        val netClient = DefaultClient(enableLogging = true)
+        assertThat(netClient.okHttpClient.interceptors, hasSize(1))
+
+        val interceptor: Interceptor = netClient.okHttpClient.interceptors[0]
+        assertThat((interceptor as HttpLoggingInterceptor).level, equalTo(HttpLoggingInterceptor.Level.BODY))
+    }
+
+    @Test
+    public fun shouldHaveDefaultTimeoutValues() {
+        val client = DefaultClient()
+        assertThat(client.okHttpClient.connectTimeoutMillis, equalTo(10 * 1000))
+        assertThat(client.okHttpClient.connectTimeoutMillis, equalTo(10 * 1000))
```

Comment:
this line should check for the **read** timeout. The same mistake is on the test below. If we make _internal_ the constant, should we use it here as well in replacement of that `10`?

Reply:
Yes, updated to check for read. I'm not using the constant because I like the test to specify expectations explicitly, not be tied to how it is implemented. Also if we do change the default, the test should fail IMO as we are changing the behavior.

Reply:
>  I like the test to specify expectations explicitly, not be tied to how it is implemented.

Yes, that's what I prefer as well. Cool 💪